tests/test_browser.py

```diff
@@ -56,7 +56,50 @@ def shell_with_script(shell_file, output_file, replacement):
     with open(output_file, 'w') as output:
       output.write(input.read().replace('{{{ SCRIPT }}}', replacement))
 
-requires_hardware = unittest.skipIf(os.environ.get("EM_LACKS_HARDWARE_ACCESS"), "This test requires hardware access including graphics and sound devices")
+def requires_hardware(maybe_hardware):
```

Comment:
this generalized version seems overly-complicated to me. how about just having `requires_graphics_hardware` and `requires_audio_hardware`, which look at `EM_LACKS_GRAPHICS_HARDWARE` etc.?

Reply:
I tried to figure out why the remaining two tests failed and now support for the old syntax is removed. Does it look better now?